core/src/main/java/org/neo4j/ogm/session/delegates/DeleteDelegate.java

```diff
@@ -50,37 +50,97 @@ public DeleteDelegate(Neo4jSession session) {
         super(session);
     }
 
-    private DeleteStatements getDeleteStatementsBasedOnType(Class type) {
-        if (session.metaData().isRelationshipEntity(type.getName())) {
-            return new RelationshipDeleteStatements();
-        }
-        return new NodeDeleteStatements();
+    public <T> void delete(T object) {
+        List<T> objectsForDeletion = createObjectsCollectionFromObject(object);
+        delete(objectsForDeletion);
     }
 
-    private <T> void deleteAll(T object) {
-        List<T> list;
-        if (object.getClass().isArray()) {
-            T[] objectsAsArray = (T[]) object;
-            list = Arrays.asList(objectsAsArray);
+    public <T> void deleteAll(Class<T> type) {
+        ClassInfo classInfo = session.metaData().classInfo(type.getName());
+        if (classInfo != null) {
+            String entityLabel = session.entityType(classInfo.name());
```

Comment:
Why don't we use `neo4jName()`here? I see that `getName()` has been used before, but that is IMHO wrong…

Reply:
This `org.neo4j.ogm.session.Neo4jSession#entityType` seems to be the method of choice in other cases.